airflow/utils/dag_processing.py

```diff
@@ -1052,18 +1052,25 @@ def prepare_file_path_queue(self):
         file_paths_recently_processed = []
         for file_path in self._file_paths:
 
+            # Find file paths that were recently processed to exclude them
+            # from being added to file_path_queue
+            # unless they were modified recently and parsing mode is "modified_time"
+            last_finish_time = self.get_last_finish_time(file_path)
             if is_mtime_mode:
                 files_with_mtime[file_path] = os.path.getmtime(file_path)
+                file_modified_time = timezone.make_aware(datetime.fromtimestamp(files_with_mtime[file_path]))
+
+                # If file is modified recently we want to process it again
+                # ignoring "self._file_process_interval" (min_file_process_interval)
+                if last_finish_time is not None and (file_modified_time < last_finish_time):
+                    file_paths_recently_processed.append(file_path)
```

Comment:
I think 4edcd5bfbbbe7cf9c251fc7e97ec3bfa722cc164 was closer.  Won't `last_finish_time` be None on a fresh scheduler start, thus never actually parse anything? Assuming we got past that, we wouldn't parse again after `file_process_interval` either, until the file is modified, right?

Reply:
I'm thinking the code block below needs to come out a level.